python/tk_multi_workfiles/file_model.py

```diff
@@ -377,7 +377,7 @@ def get_cached_file_versions(self, key, work_area, clean_only=False):
                             searching for files.
         :returns:           A dictionary {version:FileItem} of all file versions found.
         """
-        return self._search_cache.find_file_versions(work_area, key)
+        return self._search_cache.find_file_versions(work_area, key, include_dirty=not clean_only)
```

Comment:
ignoring the dirty/clean misname (I agree, I know I wrote it but it should be consistent!) this is indeed the piece that was missing :(  This was added specifically to ensure that the file save didn't decide on an incorrect version to save to if it managed for example to find all publishes before it found all work files.